sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultNetworkSpec.java

```diff
@@ -2,31 +2,55 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.mesosphere.sdk.specification.validation.ValidationUtils;
 import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 
-import javax.validation.constraints.NotNull;
-import javax.validation.constraints.Size;
+import javax.validation.Valid;
+import java.util.Map;
 
 /**
- * Default implementation of {@link NetworkSpec}.
+ * Default implementation of {@link NetworkSpec}. This class encapsulates the Container Network Interface
+ * (CNI) implementation within the SDK.
  */
 public class DefaultNetworkSpec implements NetworkSpec {
-    @NotNull
-    @Size(min = 1)
-    private String name;
+    @Valid
+    private String networkName;  // name of the overlay network to join, usually 'dcos'
+
+    @Valid
+    private Map<Integer, Integer> portMappings;  // key: host port, value: container port
 
     @JsonCreator
     public DefaultNetworkSpec(
-            @JsonProperty("name") String name) {
-        this.name = name;
+            @JsonProperty("network-name") String networkName,
+            @JsonProperty("port-mappings") Map<Integer, Integer> portMapings) {
+        this.networkName  = networkName;
+        this.portMappings = portMapings;
     }
 
-    @Override
-    public String getName() {
-        return name;
+    private DefaultNetworkSpec(Builder builder) {
+        this(
+                builder.networkName,
+                builder.portMap);
+        ValidationUtils.validate(this);
     }
 
+    public static Builder newBuilder() { return new Builder(); }
+
+    public static Builder newBuilder(NetworkSpec copy) {
+        Builder builder = new Builder();
+        builder.networkName = copy.getNetworkName();
+        builder.portMap = copy.getPortMappings();
+
+        return builder;
+    }
+
+    @Override
+    public String getNetworkName() { return networkName; }
+
+    @Override
+    public Map<Integer, Integer> getPortMappings() { return portMappings; }
```

Comment:
I know it's a one-liner, but since IDEs auto-fold anyway house style seems to recommend a fully-indented block for a method definition like this.